srtcore/srt.h

```diff
@@ -995,6 +995,20 @@ SRT_API int64_t srt_time_now(void);
 
 SRT_API int64_t srt_connection_time(SRTSOCKET sock);
 
+enum SRT_SYNC_CLOCK_TYPE
+{
+    SRT_SYNC_CLOCK_TYPE_STDCXX_STEADY      = 0, // C++11 std::chrono::steady_clock
```

Comment:
These values seem to be re-defined in sync.h. Is possible to consolidate the numeric definitions so that the two parts of the code do not have to be manually synchronized?

Reply:
The tricky thing about it is that enum values can't be used in preprocessor operators as the preprocessor does not know about the enum value.
So the following statement won't work properly
```c++
#if SRT_SYNC_CLOCK == SRT_SYNC_CLOCK_TYPE_STDCXX_STEADY
```

This is why I have additional definitions in sync.h, which I don't like as well.

I see two ways so far:
1. Change `enum SRT_SYNC_CLOCK_TYPE` in srt.h to preprocessor definitions (`#define ..`). Might look a bit ugly, although there are some already for logging.
2. Remove this enum for now, just return at integer. Also not that convenient.

Reply:
What's then exactly the point of introducing the `srt_clock_type()` function if the use of `SRT_SYNC_CLOCK` gives the same result except that the latter can be used in constant expressions?

Reply:
`SRT_SYNC_CLOCK` is an internal preprocessor definition, not exposed to API.
As requested in #1702 there should be a way to determine clock type at run time, hence the function is introduced.

Reply:
Why can't the `SRT_SYNC_CLOCK` macro be in the public API? It has its unique prefix, same as clock type symbols, they all are in the global namespace anyway, and with having all of them as macros in the public header you don't duplicate the definitions.

The only sensible alternative for that is to generate both public and private parts from a generator and keeping this as the only source of the values.